server/handlers/notify.js

```diff
@@ -1,19 +1,23 @@
-import Promise from "bluebird";
-import { smartNotifierHandler } from "hull/lib/utils";
-import updateUser from "../lib/update-user";
+const _ = require("lodash");
+const Promise = require("bluebird");
+const { smartNotifierHandler } = require("hull/lib/utils");
+
+const updateUser = require("../lib/update-user");
 
 const notify = smartNotifierHandler({
   handlers: {
     "user:update": (ctx, messages = []) => {
       const { smartNotifierResponse } = ctx;
+      const concurrency = _.get(ctx, "ship.private_settings.concurrency", 10);
       // Get 10 users every 100ms at most.
       smartNotifierResponse.setFlowControl({
         type: "next",
         size: 10,
         in: 100
       });
-      const upd = updateUser.bind(undefined, ctx);
-      return Promise.all(messages.map(upd));
+      return Promise.map(messages, (message) => {
```

Comment:
We need to apply the same concurrency limit on the batch handler too.

Reply:
It would also be useful to trace the response time for individual web hook calls for  `outgoing.user.success` and `outgoing.user.error`. And dynamically adjust the flowControl.size based on the global response time for the whole batch...